test/test_cuda.py

```diff
@@ -4148,9 +4098,68 @@ def test_no_triton_on_import(self):
             cwd=os.path.dirname(os.path.realpath(__file__))).strip().decode('ascii')
         self.assertEqual(rc, "False", "Triton was imported when importing torch!")
 
+class TestCudaOptims(TestCase):
+    # These tests will be instantiate with instantiate_device_type_tests
+    # to apply the new OptimizerInfo structure.
+
+    @onlyCUDA
+    @unittest.skipIf(not TEST_CUDA_GRAPH, "CUDA >= 11.0 or ROCM >= 5.3 required for graphs")
+    @parametrize("foreach, fused", [(False, False), (True, False), (False, True)])
+    @optims(
+        optim_db,
```

Comment:
```suggestion
        [optim for optim in optim_db if "foreach" in optim.supported_impls and "fused" in optim.supported_impls],
```
Ah, I realize you'll need this, as passing in fused or foreach flags will error on optims that don't support those.